### ISSUE TITLE
StringExtensions - isNumeric - incorrect behavior 

### ISSUE THREAD

**Issue by seifeet** [SwifterSwift/SwifterSwift#395]
### What did you do?
po "1-2".isNumeric
true

	/// SwifterSwift: Check if string contains only numbers.
	///
	///		"123".isNumeric -> true
	///		"abc".isNumeric -> false
	///

### What did you expect to happen?
Should be false

### What happened instead?
Returned true

### SwifterSwift Environment

- SwifterSwift version: 4.1
- Xcode version: 9.2
- macOS version running Xcode: High Sierra
- Swift version: 4
- Platform(s) running SwifterSwift: iOS

**Comment by SD10** [comment]
@seifeet totally agree. We should consider special characters and not just letters. Care to make a PR fixing this?

**Comment by wdcurry** [comment]
Some might consider this as a testable result. in this case, 1-2 could equal -1, which is numeric.

**Comment by seifeet** [comment]
@wdcurry yes but "1*-@.-.-2".isNumeric will also return true

**Comment by SD10** [comment]
@wdcurry You brought to my attention that we need to watch for signed numbers. 👌

**Comment by seifeet** [comment]
@SD10 yes I will create a PR fixing this!

**Comment by seifeet** [comment]
@SD10 @wdcurry would it make sense to have 2 separate functions?

isDigits - checks whether the String contains only digit characters.
isNumeric - checks whether the String a valid number.

**Comment by wdcurry** [comment]
so long as the naming is harmonious with the rest of the library for this sort of logic, sounds great.

**Comment by LucianoPAlmeida** [comment]
Hey guys :) @SD10 @seifeet @wdcurry 
I was curious about that and decided to try a regex-based approach for a fix, it took a while trying to find a simple one and finally I came up with "(^-?[\\d]+$)|(-?[\\d]+[.,]{1}[\\d]+$)"  
The implementation looks like:
```swift
public var isNumeric: Bool {
	return range(of: "(^-?[\\d]+$)|(-?[\\d]+[.,]{1}[\\d]+$)",
                     options: String.CompareOptions.regularExpression, range: nil, locale: nil) != nil
}
```
Is a option aproach to implement this fix :) 
@seifeet Your `Double("1.2") != nil`suggestion is way more simple, but maybe you want to consider this approach on your PR :))

**Comment by SD10** [comment]
I'm a pragmatic guy, I don't mind a simple cast 😅 I don't know which is more performant

**Comment by LucianoPAlmeida** [comment]
@SD10 I also don't know which one is more performant 😂 but I love regex since the day I first saw it, so my opinion on that is totally 100% biased 😂 😅

**Comment by seifeet** [comment]
@LucianoPAlmeida @SD10 I also like Regex solution but it seems like Double solution is more performant on average.
I created a simple playground to test it:
https://gist.github.com/seifeet/f860e67e41e0b90b8397096b54bb3823

**Comment by SD10** [comment]
Fixed by #396
